bios/stage-2/src/screen.rs

```diff
@@ -0,0 +1,42 @@
+use core::{arch::asm, fmt::Write};
+
+pub fn print_char(c: u8) {
+    let ax = u16::from(c) | 0x0e00;
+    unsafe {
+        asm!("push bx", "mov bx, 0", "int 0x10", "pop bx", in("ax") ax);
```

Comment:
This gives an error: "cannot use register `bx`: rbx is used internally by LLVM and cannot be used as an operand for inline asm"

Reply:
That error shouldn't occur when compiling for the bootsector target, `bx`/`rbx` are only used internally by LLVM in 32-bit and 64-bit targets.

Reply:
Ah, that could be.. Then it's probably just rust-analyzer who is complaining because it runs `cargo check` on the whole workspace. However, given that the current code works too, I'm inclined to keep it that way because it means one warning less in vscode (even if it's a false positive).

Reply:
Sure that's fine with me